### PR TITLE
deployment scripts improvements

### PR DESCRIPTION
**Pull request by hellwolf**
- allow more options when invoked as modules
- allow env `DISABLE_NATIVE_TRUFFLE` for disabling truffle native
environment when deploying

### CODE REVIEW

File: packages/ethereum-contracts/scripts/getConfig.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ const SuperfluidSDK = require("@superfluid-finance/js-sdk");
  */
 
 const DEFAULT_CONFIGS = {
-    8645: {
+    4447: {
```
Comment (pi0neerpat) [inline packages/ethereum-contracts/scripts/getConfig.js:9]: How was this port selected?

File: packages/ethereum-contracts/scripts/utils.js [hunk 2]
```diff
@@ -65,25 +65,36 @@ async function isProxiable(UUPSProxiable, address) {
...
+    }
+
+    const truffleDetected = _detectTruffle();
+    // if isTruffle is not set explicitly
+    if (!("isTruffle" in options)) {
+        if ("DISABLE_NATIVE_TRUFFLE" in process.env) {
+            options.isTruffle = !process.env.DISABLE_NATIVE_TRUFFLE;
+        } else {
+            options.isTruffle = truffleDetected;
+        }
```
Comment (pi0neerpat) [inline packages/ethereum-contracts/scripts/utils.js:97]: Starting to get so complex here

File: packages/ethereum-contracts/truffle-config.js [hunk 3]
```diff
@@ -110,7 +110,24 @@ module.exports = {
...
             network_id: 42,
             gas: 8e6,
-            gasPrice: +process.env.KOVAN_GAS_PRICE || 10e9, // 100 GWEI, goerli is busy!
+            gasPrice: +process.env.KOVAN_GAS_PRICE || 10e9,
+            //confirmations: 6, // # of confs to wait between deployments. (default: 0)
+            timeoutBlocks: 50, // # of blocks before a deployment times out  (minimum/default: 50)
+            skipDryRun: false, // Skip dry run before migrations? (default: false for public nets )
+        },
+
+        xdai: {
```
Comment (pi0neerpat) [inline packages/ethereum-contracts/truffle-config.js:119]: :+1:

**Review by pi0neerpat — CHANGES_REQUESTED** [review]

File: packages/ethereum-contracts/scripts/getConfig.js (hunk 1 above)
Comment (hellwolf) [reply packages/ethereum-contracts/scripts/getConfig.js:9]: this the local ganache chainId, i used the port number by mistake

File: packages/ethereum-contracts/scripts/utils.js (hunk 2 above)
Comment (hellwolf) [reply packages/ethereum-contracts/scripts/utils.js:97]: yea, getting

File: packages/ethereum-contracts/scripts/deploy-framework.js [hunk 4]
```diff
@@ -86,9 +97,9 @@ module.exports = async function (callback, options = {}) {
             "org.superfluid-finance.agreements.InstantDistributionAgreement.v1"
         );
 
-        newTestResolver = newTestResolver || process.env.NEW_TEST_RESOLVER;
-        useMocks = useMocks || process.env.USE_MOCKS;
-        nonUpgradable = nonUpgradable || process.env.NON_UPGRADABLE;
+        newTestResolver = newTestResolver || !!process.env.NEW_TEST_RESOLVER;
```
[the remark this answers is not in the document]
Comment (hellwolf) [reply packages/ethereum-contracts/scripts/deploy-framework.js:100]: "!!" was meant to be a shortcut to normalize those variables to booleans

File: packages/ethereum-contracts/README.md [hunk 5]
```diff
@@ -71,7 +79,8 @@ contract("My Test", accounts => {
     });
 ```
 
-To interact with the protocol, you can use the `@superfluid-finance/js-sdk`. Here is a quick-start example:
+To interact with the protocol, you can should consider to use the
```
[the remark this answers is not in the document]
Comment (hellwolf) [reply packages/ethereum-contracts/README.md]: check

File: packages/ethereum-contracts/README.md [hunk 6]
```diff
@@ -106,9 +115,45 @@ beforeEach(async () => {
...
+
+```
+...
+======== Super token deployed ========
+=============== TEST ENVIRONMENT RESOLVER ======================
+export TEST_RESOLVER_ADDRESS=0x43098b8d85Fe90eCE6B055e135759B558d2c0224
+```
+
+By exporting the `TEST_RESOLVER_ADDRESS` variable, you could then interact with your
+own deployment of the Superfluid protocol.
```
[the remark this answers is not in the document]
Comment (hellwolf) [reply packages/ethereum-contracts/README.md]: copied your version, check it out

**Review by pi0neerpat — APPROVED** [review]
:+1:

**Pull request merged**